### PR TITLE
[FEATURE] `great_expectations_contrib` CLI tool

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Introduce a CLI tool to aid contributors with custom packages
- Link to custom Cookiecutter template

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Review by abegong — APPROVED** [review]
Quick review: https://www.loom.com/share/ffbda62cce7d4b1e8b5b26ba6f285326

Nonblocking issues to consider:

* Let’s be cautious about defaults. I’ll defer to @talagluck and the DevRel team on this.
* How can we strongly incline people to use our preferred `*_expectations` naming convention?
* The cookiecutter template will need some changes and updates.
  * The directory structure needs updating
  * `requirements.dev` should have `great_expectations` as a dependency
  * Other changes to fit in with our CI + pypi deployment pipeline.

File: contrib/cli/great_expectations_contrib/cli.py
Comment (talagluck) [inline contrib/cli/great_expectations_contrib/cli.py:5]: ❤️ cookiecutter

File: contrib/cli/great_expectations_contrib/commands.py
Comment (talagluck) [inline contrib/cli/great_expectations_contrib/commands.py:86]: Silly question - is there any security risk here? Just double checking that someone couldn't accidentally publish something that they didn't mean to publish.

**Review by talagluck — APPROVED** [review]
Looks great overall! Just one question. Approving so as not to gate. Totally agree with Abe's comments, and happy to brainstorm on them, though it looks like that code is not in this PR - is that right?

File: contrib/cli/great_expectations_contrib/commands.py
Comment (kenwade4) [inline contrib/cli/great_expectations_contrib/commands.py]: Very minor, but maybe consider renaming `args` to `full_command` since the 2nd item of each Command is the name and the args combined. Or Remove the name from 2nd item (i.e. instead of `black --check .` use `--check .`) and update `run_command` to prepend the name to the list passed to `subprocess.run(`

File: contrib/cli/great_expectations_contrib/commands.py
Comment (kenwade4) [inline contrib/cli/great_expectations_contrib/commands.py]: `[name] + args.split(" ")` if you end up making the 2nd item of `Command` instances (args) not include the name

File: contrib/cli/great_expectations_contrib/commands.py
Comment (kenwade4) [inline contrib/cli/great_expectations_contrib/commands.py]: How critical is it to show the index/number next to each command being run? It just seems odd to pass in an integer to `run_command`

File: contrib/cli/great_expectations_contrib/commands.py
Comment (kenwade4) [inline contrib/cli/great_expectations_contrib/commands.py]: If you want to keep the function signature of `run_command` as it is (accepting an idx arg), you can tell `enumerate` to start counting at 1 (i.e. `enumerate(commands, 1)`) so you don't have to do `i + 1`

**Review by kenwade4 — COMMENTED** [review]
Aside from the inline comments, do we want to check that the user has a `~/.pypirc` file before proceeding? Even though `twine` is a part of the requirements, I think it will fail to publish to pypi.org if there is not a `~/.pypirc` file.

And to Abe's point about ensuring the project_name ends with "Expectations" and the project_slug ends with `_expectations`, do we want to inspect the string and append automatically if they don't have it?

File: contrib/cli/great_expectations_contrib/commands.py
Comment (cdkini) [reply contrib/cli/great_expectations_contrib/commands.py]: That's a great point! I only added `name` because there's one case with `wheel` where the first command is `python`. 

We'll want to add more checks and they won't always correspond with the first term. I'll rename for now and we can re-evaluate as we add more items.

File: contrib/cli/great_expectations_contrib/commands.py
Comment (cdkini) [reply contrib/cli/great_expectations_contrib/commands.py]: I'm going to remove the index entirely based on your point below!

File: contrib/cli/great_expectations_contrib/commands.py
Comment (cdkini) [reply contrib/cli/great_expectations_contrib/commands.py:86]: great question! perhaps we want to get rid of this and just let the user go through the steps manually? it's easy enough to remove this if need be I'm really not too sure of the security implications

File: contrib/cli/great_expectations_contrib/commands.py
Comment (cdkini) [reply contrib/cli/great_expectations_contrib/commands.py]: It is quite odd now that I'm looking back on it haha I've removed it entirely. Thanks for pointing it out!

File: contrib/cli/great_expectations_contrib/commands.py
Comment (cdkini) [reply contrib/cli/great_expectations_contrib/commands.py]: based on the response above (`name` isn't always the first value) I think we're okay with this. I'll rename the vars to make it a bit clearer!

**Review by kenwade4 — APPROVED** [review]

**Comment by cdkini** [comment]
@kenwade4 sorry just seeing your comments outside of the code.

I'm not too familiar with packaging so if you think `.pypirc` is the way to go, we can go that route! I've tested this with TestPyPi and it works without it but I'm happy to update.

I think I'll bring up this command during arch review to discuss the possible security implications @talagluck brought up. In regards to the naming conventions, Cookiecutter has input validation so we can add custom logic to reject a user's project if they don't append Expectations.

**Comment by kenwade4** [comment]
If twine will prompt the user for their pypi creds if there is no pypirc file, that's probably good enough.

**Pull request merged**
